[Search] Add Inference endpoint Flyout in Inference Management UI

Samiul-TheSoccerFan opened the pull request:
## Summary

This PR includes
- Create a UI component package to share AI connector form between multiple plugins
- Integrate the packaged components into the `Search Inference Endpoint` plugin.


https://github.com/user-attachments/assets/2b447b44-3d1d-4422-b76d-8d8fd160b2bc


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [X] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [X] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [X] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed





<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

TattdCodeMonkey commented on packages/kbn-shared-gen-ai-ui/README.md:
```diff
@@ -0,0 +1,3 @@
+# @kbn/shared-gen-ai-ui
```
If possible I would put this package in `x-pack/packages` instead of the root packages since its only going to be used by `x-pack/plugins` is I understand correctly

TattdCodeMonkey commented on packages/kbn-shared-gen-ai-ui/components/providers/get_providers.ts:
```diff
@@ -0,0 +1,42 @@
+import { INTERNAL_BASE_GEN_AI_API_PATH } from '../../constants';
+
+export const getProviders = async (http: HttpSetup): Promise<InferenceProvider[]> => {
+  return await http.get(`${INTERNAL_BASE_GEN_AI_API_PATH}/_inference/_services`);
```
Do we want to share the same API endpoints between plugins that use this package?

Abstracting the hooks with API calls will create a direct dependency on the `stack_connectors` plugin for any plugin that needs to use this package.

Did we consider abstracting just the components first?

The `search_inference_endpoints` plugin that will be using this package is used in both stack and serverless search.

Samiul-TheSoccerFan replied:
Discussed with @TattdCodeMonkey, and for now, the plan is to have the UI components focus solely on rendering, without handling any logic or making API calls. Initially, I assumed that packages could handle API calls, but it seems the best approach would be to create `lib` functions that can be shared between plugins instead.
cc: @YulNaumenko

YulNaumenko commented on .github/CODEOWNERS:
```diff
@@ -370,6 +370,7 @@ packages/kbn-ftr-common-functional-services @elastic/kibana-operations @elastic/
 packages/kbn-ftr-common-functional-ui-services @elastic/appex-qa
 packages/kbn-ftr-screenshot-filename @elastic/kibana-operations @elastic/appex-qa
 packages/kbn-gen-ai-functional-testing @elastic/appex-ai-infra
+packages/kbn-genai-common @elastic/response-ops
```
Regarding the naming, I'd recommend to have a dedicated name like `kbn-inference-endpoint-ui-common`, or something similar, because we already have a set of genai related packages and plugins, and also x-pack/packages/ai-infra/inference-common ...
As for the code owners, we should extend the list with @elastic/appex-ai-infra, @elastic/obs-ai-assistant @elastic/security-generative-ai

pgayvallet replied:
Agreed, we have already too many "gen-ai-common" things, unfortunately we probably need to be slightly more specific here.

Samiul-TheSoccerFan replied:
@YulNaumenko: Should we update the owners to only `ml-ui` as they will own both `inference management ui` and `AI connector`?

elasticmachine commented:
Pinging @elastic/kibana-core (Team:Core)

TinaHeiligers reviewed (approved):
Change to the plugin's manifest LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262062)
* Commit: 8ed537aefa7a80418e479398ba52402eaf7870b8

### Failed CI Steps
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/262062#0193dc02-9a8e-4449-821e-19ea43911edb)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/262062#0193dc27-1ced-44be-b965-221e8f36092f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/262062#0193dc02-9a8e-4449-821e-19ea43911edb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/262062/jobs/0193dc02-9a8e-4449-821e-19ea43911edb/artifacts/0193dc26-f5a2-4230-8e63-21ce50f9ee35) Jest Tests #<span></span>9 / AddInferenceFlyout valid submission
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/262062#0193dc27-1ced-44be-b965-221e8f36092f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/262062/jobs/0193dc27-1ced-44be-b965-221e8f36092f/artifacts/0193dc2c-a58e-4406-a3cc-e45611f03729) Jest Tests #<span></span>9 / AddInferenceFlyout valid submission

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262062}
buildkite-pr-comment-->

YulNaumenko commented on x-pack/packages/kbn-inference-endpoint-ui-common/src/components/additional_options_fields.tsx:
```diff
@@ -0,0 +1,292 @@
+  onSetProviderConfigEntry: (key: string, value: unknown) => Promise<void>;
+  onTaskTypeOptionsSelect: (taskType: string, provider?: string) => void;
+  selectedTaskType?: string;
+  taskTypeFormFields: ConfigEntryView[];
```
`taskTypeFormFields` is not used any more. Could you please clean up this.

YulNaumenko commented on x-pack/packages/kbn-inference-endpoint-ui-common/src/components/inference_services.tsx:
```diff
@@ -0,0 +1,359 @@
+  const [providerSchema, setProviderSchema] = useState<ConfigEntryView[]>([]);
+  const [taskTypeOptions, setTaskTypeOptions] = useState<TaskTypeOption[]>([]);
+  const [selectedTaskType, setSelectedTaskType] = useState<string>(DEFAULT_TASK_TYPE);
+  const [taskTypeFormFields] = useState<ConfigEntryView[]>([]);
```
After cleaning up taskTypeFormFields in AdditionalOptionsFields, this state also could be removed as not used.

YulNaumenko commented on x-pack/packages/kbn-inference-endpoint-ui-common/src/components/inference_services.tsx:
```diff
@@ -0,0 +1,359 @@
+  providers: InferenceProvider[];
+}
+
+export const InferenceServices: React.FC<InferenceServicesProps> = ({ providers }) => {
```
Maybe a bit more clear name could be `InferenceServiceForm` or `InferenceServiceFormFields`

YulNaumenko reviewed (approved):
LGTM! Thank you for making the shared package, awesome work! I have a draft PR where AI connector fully migrated to use a new shared components https://github.com/elastic/kibana/pull/204885 
I left a few minor clean up comments, but it could be done as a follow up.

gsoldevila commented on x-pack/packages/kbn-inference-endpoint-ui-common/kibana.jsonc:
```diff
@@ -0,0 +1,12 @@
+  ],
+  "group": "platform",
+  "visibility": "shared"
+}
```
If this `package` is going to be used from both `Search` solution and connectors, then I believe it makes sense to categorise it as _platform/shared_ like you did.

However, the package should be placed under `x-pack/platform/packages/shared`.
Can you relocate it and update the PR?

Samiul-TheSoccerFan replied:
@gsoldevila : moved the package into `x-pack/platform/packages/shared`

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262521)
* Commit: 75d8d13629ffc4fd362603eecbd562abecdfac29

### Failed CI Steps
* [Jest Tests #13](https://buildkite.com/elastic/kibana-pull-request/builds/262521#0193e061-f2ee-4562-b6b5-7241ee0a14c1)
* [FTR Configs #107](https://buildkite.com/elastic/kibana-pull-request/builds/262521#0193e061-f38a-4cbe-b3fa-d31cdc4b3453)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/262521#0193e061-f2ee-4562-b6b5-7241ee0a14c1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/262521/jobs/0193e061-f2ee-4562-b6b5-7241ee0a14c1/artifacts/0193e08a-15a5-4c8b-b466-bfa66ce8a0d1) Jest Tests #<span></span>13 / AddInferenceFlyout valid submission
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/262521#0193e061-f38a-4cbe-b3fa-d31cdc4b3453) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/262521/jobs/0193e061-f38a-4cbe-b3fa-d31cdc4b3453/artifacts/0193e091-29fb-4258-a29e-d9f656d1f932) FTR Configs #<span></span>107 / Screenshots - serverless security UI response ops docs security cases list view cases list screenshot

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 163 | 264 | +101 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 30 | +30 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 99.7KB | 219.2KB | +119.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 6.3KB | 7.0KB | +672.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 30 | +30 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 5 | +5 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/75d8d13629ffc4fd362603eecbd562abecdfac29) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 5 | +5 |


</details>


### History
* :broken_heart: [Build #262062](https://buildkite.com/elastic/kibana-pull-request/builds/262062) failed 8ed537aefa7a80418e479398ba52402eaf7870b8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262521","number":262521,"commit":"75d8d13629ffc4fd362603eecbd562abecdfac29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262521}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262564)
* Commit: 77bfeb232f110aa5fdf46b9db50a292443cdbb3b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 163 | 264 | +101 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 30 | +30 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 99.7KB | 219.4KB | +119.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `searchInferenceEndpoints` | 6.3KB | 7.0KB | +672.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 30 | +30 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 5 | +5 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/db56f019001ae98bc251798f1d8ab1e5511e8bb4) | [after](https://github.com/elastic/kibana/commit/77bfeb232f110aa5fdf46b9db50a292443cdbb3b) | diff |
| --- | --- | --- | --- |
| `@kbn/inference-endpoint-ui-common` | - | 5 | +5 |


</details>


### History
* :yellow_heart: [Build #262521](https://buildkite.com/elastic/kibana-pull-request/builds/262521) was flaky 75d8d13629ffc4fd362603eecbd562abecdfac29
* :broken_heart: [Build #262062](https://buildkite.com/elastic/kibana-pull-request/builds/262062) failed 8ed537aefa7a80418e479398ba52402eaf7870b8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262564","number":262564,"commit":"77bfeb232f110aa5fdf46b9db50a292443cdbb3b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262521","number":262521,"commit":"75d8d13629ffc4fd362603eecbd562abecdfac29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262564}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262671)
* Commit: 2dfdb4ecab8accfedc3827c90565864092585cb0
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-203204-2dfdb4ecab8a`

### History
* :green_heart: [Build #262564](https://buildkite.com/elastic/kibana-pull-request/builds/262564) succeeded 77bfeb232f110aa5fdf46b9db50a292443cdbb3b
* :yellow_heart: [Build #262521](https://buildkite.com/elastic/kibana-pull-request/builds/262521) was flaky 75d8d13629ffc4fd362603eecbd562abecdfac29
* :broken_heart: [Build #262062](https://buildkite.com/elastic/kibana-pull-request/builds/262062) failed 8ed537aefa7a80418e479398ba52402eaf7870b8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262671","number":262671,"commit":"2dfdb4ecab8accfedc3827c90565864092585cb0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262564","number":262564,"commit":"77bfeb232f110aa5fdf46b9db50a292443cdbb3b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262521","number":262521,"commit":"75d8d13629ffc4fd362603eecbd562abecdfac29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262671}
buildkite-pr-comment-->

Samiul-TheSoccerFan commented:
@elasticmachine merge upstream

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262790)
* Commit: 540ddf217dd4a52b4a8daaf43f7fd18191abc4da
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-203204-540ddf217dd4`

### History
* :green_heart: [Build #262671](https://buildkite.com/elastic/kibana-pull-request/builds/262671) succeeded 2dfdb4ecab8accfedc3827c90565864092585cb0
* :green_heart: [Build #262564](https://buildkite.com/elastic/kibana-pull-request/builds/262564) succeeded 77bfeb232f110aa5fdf46b9db50a292443cdbb3b
* :yellow_heart: [Build #262521](https://buildkite.com/elastic/kibana-pull-request/builds/262521) was flaky 75d8d13629ffc4fd362603eecbd562abecdfac29
* :broken_heart: [Build #262062](https://buildkite.com/elastic/kibana-pull-request/builds/262062) failed 8ed537aefa7a80418e479398ba52402eaf7870b8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262790","number":262790,"commit":"540ddf217dd4a52b4a8daaf43f7fd18191abc4da"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262671","number":262671,"commit":"2dfdb4ecab8accfedc3827c90565864092585cb0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262564","number":262564,"commit":"77bfeb232f110aa5fdf46b9db50a292443cdbb3b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262521","number":262521,"commit":"75d8d13629ffc4fd362603eecbd562abecdfac29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262790}
buildkite-pr-comment-->

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#7629](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7629)
[❌] x-pack/test_serverless/functional/test_suites/search/config.ts: 22/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F203204%2Fhead)

efegurkan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx:
```diff
@@ -0,0 +1,357 @@
+        key: k,
+        isValid: true,
+        ...newProvider?.configurations[k],
+      })) as ConfigEntryView[];
```
Not only this but in general, can we do this transformations without `as`? what happens if you set return type `ConfigEntryView` to the arrow function in `map`? 

currently we override whatever it is and cast the type to `ConfigEntryView[]` and missing errors to be introduced with a change.

efegurkan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/configuration/configuration_field.tsx:
```diff
@@ -0,0 +1,232 @@
+  validateAndSetConfigValue,
+}) => {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { isValid, value, default_value, key } = configEntry;
```
`default_value: defaultValue`

efegurkan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx:
```diff
@@ -0,0 +1,357 @@
+    ],
+  });
+
+  const handleProviderPopover = useCallback(() => {
```
nitpick: `toggleProviderPopover`

efegurkan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx:
```diff
@@ -0,0 +1,357 @@
+    setProviderPopoverOpen((isOpen) => !isOpen);
+  }, []);
+
+  const handleProviderClosePopover = useCallback(() => {
```
nitpick: `closeProviderPopover`

efegurkan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/inference_service_form_fields.tsx:
```diff
@@ -0,0 +1,357 @@
+          const itemValue = item;
+          itemValue.isValid = true;
+          if (item.sensitive && secrets?.providerSecrets) {
+            itemValue.value = secrets?.providerSecrets[item.key] as any;
```
can we avoid these `as any` casts ?

efegurkan reviewed (changes requested):
In general it is looking good. 

There are a few changes I have requested, others are nitpicks.

Once it is done, I will review this another time.

Samiul-TheSoccerFan commented on x-pack/plugins/search_inference_endpoints/public/components/add_inference_endpoints/add_inference_flyout_wrapper.tsx:
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+import { InferenceForm } from './inference_form';
+import * as i18n from './translations';
```
This will require a huge refactor throughout this plugin. I added this as an action item in [tech debts](https://elasticco.atlassian.net/browse/SEARCH-713).

Samiul-TheSoccerFan commented on x-pack/plugins/search_inference_endpoints/public/components/add_inference_endpoints/add_inference_flyout_wrapper.tsx:
```diff
@@ -0,0 +1,66 @@
+  return (
+    <EuiFlyout
+      ownFocus
+      onClose={() => onClose(false)}
```
Good catch

Samiul-TheSoccerFan commented on x-pack/plugins/search_inference_endpoints/server/lib/add_inference_endpoint.ts:
```diff
@@ -0,0 +1,47 @@
+      inference_id: config?.inferenceId ?? '',
+      task_type: config?.taskType as InferenceTaskType,
+      inference_config: {
+        service: config!.provider,
```
making it optional and will display the error message received from the `ES`.

Samiul-TheSoccerFan commented on x-pack/plugins/search_inference_endpoints/server/lib/add_inference_endpoint.ts:
```diff
@@ -0,0 +1,47 @@
+      ...unflattenObject(secrets?.providerSecrets ?? {}),
+    };
+
+    return await esClient?.inference.put({
```
My mistake, ES client is a required param for this function.

Samiul-TheSoccerFan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/utils/helpers.ts:
```diff
@@ -0,0 +1,80 @@
+
+export const generateInferenceEndpointId = (config: Config) => {
+  const taskTypeSuffix = config.taskType ? `${config.taskType}-` : '';
+  const inferenceEndpointId = `${config.provider}-${taskTypeSuffix}${Math.random()
```
We do send an error message if somehow the inference endpoint overlaps.

Samiul-TheSoccerFan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/configuration/configuration_utils.ts:
```diff
@@ -0,0 +1,49 @@
+
+export const ensureIntType = (value: string | number | boolean | null): number | null => {
+  // int is null-safe to prevent empty values from becoming zeroes
+  if (value === null || value === '') {
```
I played around with this function and it returns the same result ('`true`) for both inputs `''` and `'   '` Which is a correct assumption. The goal is to eliminate floating point and the server will take care of the rest of the validations.

Samiul-TheSoccerFan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/utils/helpers.ts:
```diff
@@ -0,0 +1,80 @@
+                !configData[field.key] ||
+                (typeof configData[field.key] === 'string' && isEmpty(configData[field.key]))
+              ) {
+                field.validationErrors = [i18n.getRequiredMessage(field.label)];
```
both `validationErrors` and `isValid` is part of `ConfigEntryView`.

Samiul-TheSoccerFan commented on x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/configuration/configuration_form_items.tsx:
```diff
@@ -0,0 +1,90 @@
+  isLoading: boolean;
+  items: ConfigEntryView[];
+  setConfigEntry: (key: string, value: string | number | boolean | null) => void;
+  direction?: 'column' | 'row' | 'rowReverse' | 'columnReverse' | undefined;
```
We probably do not need all. We want to provide it as an option for now and probably in the future will only keep `column` as a valid option.

Samiul-TheSoccerFan commented:
@efegurkan Addressed most of the comments. The package is kind of shared and copied from `AI connector` so I can not make a huge change at this moment as it might break `AI connector` too. I have created few tickets to address these comments(which did not resolved in this PR) later part of the project.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/262892)
* Commit: 1f99b0ba1217e872e278bac35a3a0b3ce738a69f
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-203204-1f99b0ba1217`

### History
* :green_heart: [Build #262790](https://buildkite.com/elastic/kibana-pull-request/builds/262790) succeeded 540ddf217dd4a52b4a8daaf43f7fd18191abc4da
* :green_heart: [Build #262671](https://buildkite.com/elastic/kibana-pull-request/builds/262671) succeeded 2dfdb4ecab8accfedc3827c90565864092585cb0
* :green_heart: [Build #262564](https://buildkite.com/elastic/kibana-pull-request/builds/262564) succeeded 77bfeb232f110aa5fdf46b9db50a292443cdbb3b
* :yellow_heart: [Build #262521](https://buildkite.com/elastic/kibana-pull-request/builds/262521) was flaky 75d8d13629ffc4fd362603eecbd562abecdfac29
* :broken_heart: [Build #262062](https://buildkite.com/elastic/kibana-pull-request/builds/262062) failed 8ed537aefa7a80418e479398ba52402eaf7870b8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262892","number":262892,"commit":"1f99b0ba1217e872e278bac35a3a0b3ce738a69f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262790","number":262790,"commit":"540ddf217dd4a52b4a8daaf43f7fd18191abc4da"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262671","number":262671,"commit":"2dfdb4ecab8accfedc3827c90565864092585cb0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262564","number":262564,"commit":"77bfeb232f110aa5fdf46b9db50a292443cdbb3b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262521","number":262521,"commit":"75d8d13629ffc4fd362603eecbd562abecdfac29"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/262062","number":262062,"commit":"8ed537aefa7a80418e479398ba52402eaf7870b8"}],"number":262892}
buildkite-pr-comment-->

efegurkan reviewed (approved):
Looks good.

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/12488907363

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 203204
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

Samiul-TheSoccerFan commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205171">](https://github.com/elastic/kibana/pull/205171)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->

Samiul-TheSoccerFan commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205172">](https://github.com/elastic/kibana/pull/205172)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
